maxtext_gpu_dependencies.Dockerfile

```diff
@@ -38,13 +38,12 @@ RUN mkdir -p /deps
 # Set the working directory in the container
 WORKDIR /deps
 
-# Copy necessary build files to docker container
-COPY setup.sh requirements.txt constraints_gpu.txt /deps/
```

Comment:
This is done so that when we are making changes independent of dependencies, we won't do the installation step again, which can take >2m. Is there a specific concern to overrule it? @khatwanimohit

Reply:
setup.sh now includes running the `pre-commit install` command which installs the pre-commit hooks in the user's git repo. That command needs access to `.pre-commit-config.yaml` file